tests/Doctrine/Tests/DBAL/Functional/Schema/MySqlSchemaManagerTest.php

```diff
@@ -283,6 +285,7 @@ public function testColumnCollation() : void
         self::assertEquals('latin1_swedish_ci', $columns['text']->getPlatformOption('collation'));
         self::assertEquals('latin1_swedish_ci', $columns['foo']->getPlatformOption('collation'));
         self::assertEquals('utf8_general_ci', $columns['bar']->getPlatformOption('collation'));
+        self::assertInstanceOf(BlobType::class, $columns['baz']->getType());
```

Comment:
So as far as I understand, if MySQL is told to create a `(VAR)CHAR`/`TEXT` column with the `BINARY` collation, it converts it to `(VAR)BINARY`/`BLOB` which doesn't expose its collation via the `INFORMATION_SCHEMA.COLUMNS`. Therefore, the assertion is made based on the column type.

While the approach seems legit, I'm curious what's the practical use case where you'd need to create a `BINARY` column as `CHAR COLLATION BINARY`?

Reply:
Yes. I can't say anything about the practical use case. We fixed/enabled it in Laravel and this is the port to Doctrine (which Laravel uses to alter columns).

You can argue about the use case, but I think quoting the collation is not unreasonable.

Reply:
It is reasonable.